app/models/conversation.rb

```diff
@@ -39,4 +40,26 @@ def self.share_conversation(user, match)
   def not_current_user(current_user)
     users.where.not(id: current_user.id).to_a
   end
+
+  def self.with_unread_messages_count_for(user_id)
+    Conversation
+      .select('conversations.*, conversation_users.unread_messages AS unread_count')
+      .joins(:conversation_users)
+      .where(conversation_users: { user_id: user_id })
+  end
+
+  def new_message(user_id)
+    conversations_with_new_messages =
+      conversation_users.where.not(user_id: user_id)
```

Comment:
I would move this to a scope `not_for_user(user_id)` so you can reuse if necessary.